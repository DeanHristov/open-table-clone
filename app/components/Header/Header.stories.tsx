import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Header, { IHeaderProps } from "./Header";
import Input from "../Input";
import Button from "../Button";

export default {
  title: "Components/Header",
  component: Header,
} as Meta;

export const Default: Story<IHeaderProps> = (args) => (
  <Header {...args}>
    <div className="pr-4">
      <Input placeholder="Type location..." />
    </div>
    <Button red medium title={"Let's go"} />
  </Header>
);
export const Search: Story<IHeaderProps> = (args) => (
  <Header {...args}>
    <div className="pr-4">
      <Input placeholder="Type location..." />
    </div>
    <Button red medium title={"Let's go"} />
  </Header>
);

Default.args = {
  height: "h-64",
};
Search.args = {};
