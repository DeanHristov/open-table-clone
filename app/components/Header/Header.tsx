import { FC, ReactNode } from "react";

export interface IHeaderProps {
  children: ReactNode;
  height?: string;
}

const Header: FC<IHeaderProps> = ({ children, height = "" }) => {
  return (
    <header
      className={`${height} flex justify-center items-center bg-gradient-to-r from-[#0f1f47] to-[#5f6984] py-8`}
    >
      {children}
    </header>
  );
};

export default Header;
