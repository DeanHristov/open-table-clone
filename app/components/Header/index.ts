import Header, { IHeaderProps } from "./Header";

export type { IHeaderProps };
export default Header;
