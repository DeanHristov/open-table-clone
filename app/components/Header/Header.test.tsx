import { render } from "@testing-library/react";
import Header from "./Header";

describe("Component: <Header {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(
      <Header>
        <div />
      </Header>
    );

    expect(container).toMatchSnapshot();
  });
});
