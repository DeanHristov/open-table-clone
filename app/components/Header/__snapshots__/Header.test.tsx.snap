// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Header {...} /> Should be able to render default state 1`] = `
<div>
  <header
    class=" flex justify-center items-center bg-gradient-to-r from-[#0f1f47] to-[#5f6984] py-8"
  >
    <div />
  </header>
</div>
`;
